Library/Tracking/KSRAnalyticsIdentityData.swift

```diff
@@ -0,0 +1,81 @@
+import Foundation
+import KsApi
+
+private enum KSRAnalyticsIdentityTraitValue: Equatable {
+  case string(String)
+  case bool(Bool)
+
+  var value: Any {
+    switch self {
+    case let .string(value): return value
+    case let .bool(value): return value
+    }
+  }
+}
+
+public struct KSRAnalyticsIdentityData: Equatable {
+  public let userId: Int
+  private let name: String
+  private let notifications: User.Notifications
+
+  init(_ user: User) {
+    self.userId = user.id
+    self.name = user.name
+    self.notifications = user.notifications
+  }
+
+  public static func == (lhs: KSRAnalyticsIdentityData, rhs: KSRAnalyticsIdentityData) -> Bool {
+    let uniqueTraits = lhs.uniqueTraits(comparedTo: rhs)
+
+    return uniqueTraits.isEmpty
+  }
+
+  func uniqueTraits(comparedTo otherData: KSRAnalyticsIdentityData?) -> [String: Any] {
+    var newTraits: [String: Any] = [:]
+    let otherTraits = otherData?.traits ?? [:]
+
+    for key in self.traits.keys where self.traits[key] != otherTraits[key] {
+      newTraits[key] = self.traits[key]?.value
+    }
+
+    return newTraits
+  }
+
+  fileprivate var traits: [String: KSRAnalyticsIdentityTraitValue] {
+    let notifications = self.notifications.encode()
+      .mapValues { ($0 as? Bool).flatMap(KSRAnalyticsIdentityTraitValue.bool) }
+      .compactMapValues { $0 }
+
+    return [
+      "name": .string(self.name)
+    ]
+    .withAllValuesFrom(notifications)
+  }
+}
+
+extension KSRAnalyticsIdentityData: Codable {
+  private enum CodingKeys: String, CodingKey {
+    case userId
+    case name
+    case notifications
+  }
+
+  public func encode(to encoder: Encoder) throws {
```

Comment:
what codification do you use?

Reply:
Sorry, not sure what you mean? This is using `Swift`'s `Codable` protocol conformance so that we can serialize this model into a `Data` object and store it in our key value store.